src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -163,14 +163,18 @@ internal static void AdvanceToPosition(this Stream stream, long position)
         {
             long numBytesLeft = position - stream.Position;
             Debug.Assert(numBytesLeft >= 0);
-            while (numBytesLeft != 0)
+            if (numBytesLeft > 0)
             {
                 const int throwAwayBufferSize = 64;
```

Comment:
Nit: if you make any more edits here, this const isn't really needed... you can just use throwAwayBuffer.Length where it's used beyond here.  throwAwayBuffer can also be renamed to just buffer or something like that.